From 24a4c9abc2c4c6d0aa84f9045424f62b8b9bdbe9 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Tue, 5 Mar 2024 16:59:05 +0100
Subject: [PATCH] ipu6-isys: Add error logging to various probe error paths

Add error logging to various probe() error paths to make it easier to
debug probe() failures.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 drivers/media/pci/intel/ipu-isys-csi2.c   |  4 +++-
 drivers/media/pci/intel/ipu-isys-subdev.c |  9 +++++++--
 drivers/media/pci/intel/ipu-isys.c        | 16 ++++++++++++----
 3 files changed, 22 insertions(+), 7 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-csi2.c b/drivers/media/pci/intel/ipu-isys-csi2.c
index 2dad9ce7dfcc..4c47de9169a9 100644
--- a/drivers/media/pci/intel/ipu-isys-csi2.c
+++ b/drivers/media/pci/intel/ipu-isys-csi2.c
@@ -531,8 +531,10 @@ int ipu_isys_csi2_init(struct ipu_isys_csi2 *csi2,
 				    NR_OF_CSI2_SOURCE_PADS,
 				    NR_OF_CSI2_SINK_PADS,
 				    0);
-	if (rval)
+	if (rval) {
+		dev_err(&isys->adev->dev, "ipu_isys_subdev_init() err %d\n", rval);
 		goto fail;
+	}
 
 	csi2->asd.pad[CSI2_PAD_SINK].flags = MEDIA_PAD_FL_SINK
 		| MEDIA_PAD_FL_MUST_CONNECT;
diff --git a/drivers/media/pci/intel/ipu-isys-subdev.c b/drivers/media/pci/intel/ipu-isys-subdev.c
index b9f48d41f844..6c0180d50ab1 100644
--- a/drivers/media/pci/intel/ipu-isys-subdev.c
+++ b/drivers/media/pci/intel/ipu-isys-subdev.c
@@ -819,17 +819,22 @@ int ipu_isys_subdev_init(struct ipu_isys_subdev *asd,
 		return -ENOMEM;
 
 	rval = media_entity_pads_init(&asd->sd.entity, num_pads, asd->pad);
-	if (rval)
+	if (rval) {
+		dev_err(&asd->isys->adev->dev, "%s: media_entity_pads_init() err %d\n", __func__, rval);
 		goto out_mutex_destroy;
+	}
 
 	if (asd->ctrl_init) {
 		rval = v4l2_ctrl_handler_init(&asd->ctrl_handler, nr_ctrls);
-		if (rval)
+		if (rval) {
+			dev_err(&asd->isys->adev->dev, "%s: v4l2_ctrl_handler_init() err %d\n", __func__, rval);
 			goto out_media_entity_cleanup;
+		}
 
 		asd->ctrl_init(&asd->sd);
 		if (asd->ctrl_handler.error) {
 			rval = asd->ctrl_handler.error;
+			dev_err(&asd->isys->adev->dev, "%s: ctrl_handler.error %d\n", __func__, rval);
 			goto out_v4l2_ctrl_handler_free;
 		}
 
diff --git a/drivers/media/pci/intel/ipu-isys.c b/drivers/media/pci/intel/ipu-isys.c
index 3cc38d4c959a..19b3fee56dd2 100644
--- a/drivers/media/pci/intel/ipu-isys.c
+++ b/drivers/media/pci/intel/ipu-isys.c
@@ -396,8 +396,10 @@ static int isys_register_subdevices(struct ipu_isys *isys)
 		rval = ipu_isys_csi2_init(&isys->csi2[i], isys,
 					  isys->pdata->base +
 					  csi2->offsets[i], i);
-		if (rval)
+		if (rval) {
+			dev_err(&isys->adev->dev, "ipu_isys_csi2_init() err %d\n", rval);
 			goto fail;
+		}
 
 		isys->isr_csi2_bits |= IPU_ISYS_UNISPART_IRQ_CSI2(i);
 	}
@@ -1009,12 +1011,16 @@ static int isys_register_devices(struct ipu_isys *isys)
 		goto out_v4l2_device_unregister;
 
 	rval = isys_notifier_init(isys);
-	if (rval)
+	if (rval) {
+		dev_err(&isys->adev->dev, "isys_notifier_init() err %d\n", rval);
 		goto out_isys_unregister_subdevices;
+	}
 
 	rval = v4l2_device_register_subdev_nodes(&isys->v4l2_dev);
-	if (rval)
+	if (rval) {
+		dev_err(&isys->adev->dev, "error registering subdev nodes %d\n", rval);
 		goto out_isys_notifier_cleanup;
+	}
 
 	return 0;
 
@@ -1561,8 +1567,10 @@ static int isys_probe(struct ipu_bus_device *adev)
 	if (rval)
 		goto out_remove_pkg_dir_shared_buffer;
 	rval = isys_iwake_watermark_init(isys);
-	if (rval)
+	if (rval) {
+		dev_err(&adev->dev, "isys_iwake_watermark_init() err %d\n", rval);
 		goto out_unregister_devices;
+	}
 
 	ipu_mmu_hw_cleanup(adev->mmu);
 
